scripts/create_burndown.py

```diff
@@ -17,211 +17,402 @@
+    	       xytext=(days[0], open_tasks[0] + 0.5), textcoords='data',
+    	       arrowprops=dict(arrowstyle="<|-|>", connectionstyle="arc3", color=color)
+    	      )
+    
+      y_text = scalefactor * max_story_points
+      self.t.text(0.7, y_text, str(int(tasks_done)) + " tasks done",
+    	   rotation='vertical', verticalalignment='top', color=color
+    	  )
+    
+    if bonus_tasks_day_one:
+      self.t.annotate("",
+    	       xy=(days[0], scalefactor * bonus_tasks_day_one - 0.5 ), xycoords='data',
+    	       xytext=(days[0], 0.5 - bonus_tasks_day_one), textcoords='data',
+    	       arrowprops=dict(arrowstyle="<|-|>", connectionstyle="arc3", color=color)
+    	      )
+    
+      y_text = - bonus_tasks_done[0] / 2
+      self.t.text(0.4, y_text, str(int(bonus_tasks_done[0])) + " extra",
+    	   rotation='vertical', verticalalignment='center', color=color
+    	  )
+      self.t.text(0.7, y_text, "tasks done",
+    	   rotation='vertical', verticalalignment='center', color=color
+    	  )
+    return
+
+  # Set the parameters for the fast lane graph
+  def setFastLaneConfig(self, label, color, ymin, ymax, fig):
+    self.fl = self.sp.twinx()
+    self.fl.set_ylabel(label, color=color)
+    self.fl.tick_params(axis='y', colors=color)
+    self.fl.set_ylim([ymin, ymax])
+    self.fl.spines['right'].set_position(('axes', 1.15))
+    fig.subplots_adjust(right=0.8)
+    return
+
+  # Plot the fast lane graph
+  def plotFastLane(self, x_fast_lane, y_fast_lane, linestyle, linewidth):
+    plt.plot(x_fast_lane, y_fast_lane, linestyle, linewidth=linewidth)
+    return
+  
+  # Calculate and plot new fast lane cards as a bar at the corresponding day
+  def plotNewFastLane(self, total_fast_lane, width, color):
+    if len(total_fast_lane) > 1:
+      new_fast_lane_cards = [0]
+      for i in range(1, len(total_fast_lane)):
+        new_fast_lane_cards.append(total_fast_lane[i] - total_fast_lane[i - 1])
+      effective_new_fast_lane_days = []
+      effective_new_fast_lane_cards = []
+      for i in range(len(new_fast_lane_cards)):
+        if new_fast_lane_cards[i] > 0:
+          effective_new_fast_lane_days.append(i + 0.1 + 1)
+          effective_new_fast_lane_cards.append(new_fast_lane_cards[i])
+      if len(effective_new_fast_lane_cards) > 0:
+        plt.bar(effective_new_fast_lane_days, effective_new_fast_lane_cards, width, color=color)
+    return
+  
+  
 # Save the burndown chart
-plt.savefig('burndown-' + args.sprint + '.png',bbox_inches='tight')
-if not args.no_head:
-  plt.show()
+def saveBurndown(args):
+  plt.savefig('burndown-' + args.sprint + '.png',bbox_inches='tight')
+  if not args.no_head:
+    plt.show()
+
+  return
+
+
+### MAIN ###
+
+# Parse command arguments and/or print help
+args = argParser()
+  
+# Create burndown data object
+bd = BurndownData(args)
+
+# Configure plot parameters
+pc = PlotConfig(bd)
+
+# Plot all graphs
+graph = Graph(bd, args, pc)
+
+# Save the burndown chart
+saveBurndown(args)
```

Comment:
In this part the actual control flow of creating the burndown chart takes place. It would be clearer, if it would more work like shown in the examples above.

So instead of calling one function after the other and forwarding return values as parameters it could create the relevant objects and call their methods to get the data, build up the Plot, and draw the individual elements.